### ISSUE TITLE
Current README leaves out instructions for getting framework submodule code

### ISSUE THREAD

**Issue by gsrohde** [biocro/skelBML#2]
As is, following the README instructions will yield an installation error about missing files:

```
% R CMD INSTALL skelBML
* installing to library ‘/Users/srohde/Library/R/x86_64/4.2/library’
* installing *source* package ‘skelBML’ ...
** using staged installation
** libs
clang++ -mmacosx-version-min=10.13 -std=gnu++11 -MT R_module_library.o -MMD -MP -MF .deps/R_module_library.d -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I../boost_1_71_0 -DR_NO_REMAP  -I/usr/local/include   -fPIC  -Wall -g -O2  -c -o R_module_library.o R_module_library.cpp
R_module_library.cpp:5:10: fatal error: 'framework/state_map.h' file not found
#include "framework/state_map.h"           // for string_vector
         ^~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make: *** [R_module_library.o] Error 1
ERROR: compilation failed for package ‘skelBML’
```

**Comment by gsrohde** [comment]
As is, following the README instructions will yield an installation error about missing files:

```
% R CMD INSTALL skelBML
* installing to library ‘/Users/srohde/Library/R/x86_64/4.2/library’
* installing *source* package ‘skelBML’ ...
** using staged installation
** libs
clang++ -mmacosx-version-min=10.13 -std=gnu++11 -MT R_module_library.o -MMD -MP -MF .deps/R_module_library.d -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I../boost_1_71_0 -DR_NO_REMAP  -I/usr/local/include   -fPIC  -Wall -g -O2  -c -o R_module_library.o R_module_library.cpp
R_module_library.cpp:5:10: fatal error: 'framework/state_map.h' file not found
#include "framework/state_map.h"           // for string_vector
         ^~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make: *** [R_module_library.o] Error 1
ERROR: compilation failed for package ‘skelBML’
```

**Comment by eloch216** [comment]
How did you obtain your local copy of the files? I use GitHub desktop, and the submodule files are automatically included. I didn't realize that if I just download the code as a zip archive from the GitHub website, it won't include the submodule files. I'm not sure what the instructions would be in this case, other than just downloading the framework files separately and putting them in the correct folder.

**Comment by eloch216** [comment]
Seems like there might not be a way to include the submodule files when downloading directly from the GitHub website: https://stackoverflow.com/questions/12936014/include-a-github-submodule-automatically-in-download

**Comment by eloch216** [comment]
I think if you clone the repository directly using git, it can be done properly with `git clone --recursive https://github.com/biocro/skelBML.git`. I haven't tried it yet, though.

**Comment by gsrohde** [comment]
> I think if you clone the repository directly using git, it can be done properly with `git clone --recursive https://github.com/biocro/skelBML.git`. I haven't tried it yet, though.

Yes, that should work.

But since I usually copy the clone command to my clipboard by clicking on the GitHub clone icon, it also works to follow the clone with
```
git submodule init
git submodule update
```
or, in one command,
```
git submodule update --init
```

**Comment by gsrohde** [comment]
> I didn't realize that if I just download the code as a zip archive from the GitHub website, it won't include the submodule files. I'm not sure what the instructions would be in this case, other than just downloading the framework files separately and putting them in the correct folder.

I thought you said at the meeting yesterday that users shouldn't do this anyway, since if they do, they can't easily get updates to the `skelBML` repository.

**Comment by gsrohde** [comment]
I'm tabling this until decisions pertaining to PR #1 are resolved.

**Comment by eloch216** [comment]
I think this issue was addressed by this commit, which has been merged into `main`: https://github.com/biocro/skelBML/commit/64ad0ff46b80a891d01f071611f667265a535542

@gsrohde let me know if you think the new instructions are sufficient.

**Comment by gsrohde** [comment]
I made a short PR (#11) to tweak the instructions a bit.  Specifically:
* I made mention of renaming the repository to a suitable library name in step 1 so that it doesn't come as a surprise when step 3 talks about choosing a library name and says "ideally this would be the name of the new forked repository."  (One _can_ rename the fork at this point, but it is simpler to give it the desired name in the first place.)
* For someone familiar with using Git on the command line and _not_ familiar with using GitHub Desktop, it is arguably easier to run the two needed Git commands on the command line than it is to install GitHub Desktop and figure out its interface.  So I tried to put these two options on a more equal footing.  I also tried to make the instructions in step 2 more specific.  It wasn't immediately clear in your instructions, for example, whether the local copy should be obtained from `biocro/skelBML` or from the fork since the instructions just say 'use the "Open with GitHub Desktop" option from the GitHub website.'
* I eliminated mention of the "Download Zip" option, since it seemed unnecessary and distracting.
* [I perhaps should trademark _GutHub_ before someone else does!]
